Source/SourceKittenFramework/String+SourceKitten.swift

```diff
@@ -232,14 +232,15 @@ extension NSString {
     - parameter characterSet: Character set to check for membership.
     */
     public func trimmingTrailingCharacters(in characterSet: CharacterSet) -> String {
-        if length == 0 {
+        guard length > 0 else {
             return ""
         }
-        var charBuffer = [unichar](repeating: 0, count: length)
-        getCharacters(&charBuffer, range: NSRange(location: 0, length: charBuffer.count))
-        for newLength in (1...length).reversed() {
-            if !characterSet.contains(UnicodeScalar(charBuffer[newLength - 1])!) {
-                return substring(with: NSRange(location: 0, length: newLength))
+        var string = self.bridge()
+        let endIndex = string.unicodeScalars.endIndex
+        for offset in 1...string.unicodeScalars.count {
+            let index = string.unicodeScalars.index(endIndex, offsetBy: -offset)
```

Comment:
This creates the index from the end of the string every time.
Can you reuse the index created in the previous loop to reduce computational cost?